src/avm2/native.ts

```diff
@@ -1063,14 +1063,17 @@ module Shumway.AVM2.AS {
       if (re === (void 0) || re === null) {
         return null;
       } else {
-        if (re instanceof XRegExp && re.global) {
-          var matches = [], m;
-          while ((m = re.exec(this))) {
-            matches.push(m[0]);
+        if (re instanceof ASRegExp) {
```

Comment:
(Well, testing this it seems like the `instanceof` actually works. I don't know why it would, but great. Still, as a general rule always use at least the last part of the package path in situations like this.)